packages/osc-ui/src/components/Carousel/Carousel.tsx

```diff
@@ -1,274 +1,368 @@
+    };
+    const keenPlugins = () => {
+        let plugins: KeenSliderPlugin<{}, {}, KeenSliderHooks>[] = [];
+        plugins.push(AdaptiveHeight);
+        if (parseInt(delay) > 0) {
+            plugins.push(AutoPlay);
+        }
+        plugins.push(ResizePlugin);
+        return plugins;
+    };
 
-    if (typeof document !== 'undefined') {
-        let r = document.querySelector(`.embla__carousel_wrapper_${carouselKey}`) as any;
-        if (r) {
-            if (slidesPerPage) {
-                r.style.setProperty(
-                    '--embla__slidesPerPage',
-                    `calc(${(1 / slidesPerPage) * 100}%)`
-                );
+    const newSlidesInView = (slider) => {
+        const loadedSlides = slider.track.details.slides.map((slide, idx) => {
+            if (slide.portion > 0) {
+                return true;
+            } else {
+                return slidesInView[idx];
             }
-            if (slideGap) {
-                r.style.setProperty('--embla__slideGap', slideGap + 'px');
+        });
+        console.log('this is the old slides in view', slidesInView);
+        console.log('this is the new slides in view', loadedSlides);
+        return loadedSlides;
+    };
+
+    const newAriaHidden = (slider) => {
+        const loadedSlides = slider.track.details.slides.map((slide, idx) => {
+            if (slide.portion > 0) {
+                return true;
+            } else {
+                return false;
             }
-            if (axis) {
-                r.style.setProperty('--embla__axis', axis === 'x' ? 'row' : 'column');
+        });
+        console.log('this is the old slides in view', slidesInView);
+        console.log('this is the new slides in view', loadedSlides);
+        return loadedSlides;
+    };
+
+    const [currentSlide, setCurrentSlide] = React.useState(startPage - 1);
+    const [slidesInView, setSlidesInView] = React.useState<boolean[]>(
+        React.Children.toArray(children).map((_, idx) => {
+            if (between(idx, startPage - 1, startPage - 1 + trueSlidesPerPage)) {
+                return true;
             }
-            if (height) {
-                r.style.setProperty('--embla__height', height + 'px');
-                r.style.setProperty(
-                    '--embla__style_height',
-                    `calc(${parseInt(height, 10) / slidesPerPage}px - ${slideGap * 2}px)`
-                );
+            return false;
+        })
+    );
+    const [ariaHidden, setAriaHidden] = React.useState<boolean[]>(
+        React.Children.toArray(children).map((_, idx) => {
+            if (between(idx, startPage - 1, startPage - 1 + trueSlidesPerPage)) {
+                return true;
             }
-        }
-    }
+            return false;
+        })
+    );
+    const [loaded, setLoaded] = useState(false);
+    const [sliderRef, instanceRef] = useKeenSlider<HTMLDivElement>(
+        {
+            loop,
+            drag: indicatorActive || dotsActive,
```

Comment:
This should be true by default, we would always expect the slider to be dragable
```suggestion
            drag: true,
```